### PR TITLE
[IMPROVEMENT] Refactor Analytics Events

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_

As part of the Segment integration, it's required to do a series of refactors on the app analytics system. The PR https://github.com/MetaMask/metamask-mobile/pull/5139 is the main migration development, but getting the events refactor merged will help to reduce the number of code conflicts.

_2. What is the improvement/solution?_

Refactor the analytics events and centralize them into a single file `app/core/Analytics/MetaMetrics.events.ts`

**Screenshots/Recordings**

~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/372

Related to PR https://github.com/MetaMask/metamask-mobile/pull/5139

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/core/Analytics/MetaMetrics.events.ts [hunk 1]
```diff
@@ -5,19 +5,1070 @@
...
+  action?: ACTIONS,
+  description?: DESCRIPTION,
+): IMetaMetricsEvent => {
+  if (action && description) {
+    return { name, properties: { action, name: description } };
+  }
+  return { name };
+};
+
+export const ONBOARDING_WIZARD_STEP_DESCRIPTION = {
```
Comment (jpcloureiro) [inline app/core/Analytics/MetaMetrics.events.ts:19]: remove export statement here & add `ONBOARDING_WIZARD_STEP_DESCRIPTION` to the export statement on the bottom

File: app/core/Analytics/MetaMetrics.events.ts [hunk 2]
```diff
@@ -5,19 +5,1070 @@
...
+  3: 'Account Name',
+  4: 'Main Navigation',
+  5: 'Browser',
+  6: 'Search',
+};
+
+/**
+ * V2 Analytics Tracking Events
+ */
+export enum EVENT_NAME {
```
Comment (jpcloureiro) [inline app/core/Analytics/MetaMetrics.events.ts:31]: remove export statement here & add `EVENT_NAME` to the export statement on the bottom

File: app/util/analytics.js [hunk 3]
```diff
@@ -1,623 +0,0 @@
-export const ONBOARDING_WIZARD_STEP_DESCRIPTION = {
```
Comment (jpcloureiro) [inline app/util/analytics.js:1]: Can we remove this file from the project?

**Review by owencraston — APPROVED** [review]
I didn't run the app but all the events seem to match. 
Unrelated but I created this util for [generateDeviceAnalyticsMetaData](https://github.com/MetaMask/metamask-mobile/blob/main/app/util/metrics/DeviceAnalyticsMetaData/generateDeviceAnalyticsMetaData.ts). Perhaps this is something we want to move in future changes or even make it a part of all the events? Just thought i'd mention it here to create the paper trail.  Nice work 👍

**Review by jpcloureiro — CHANGES_REQUESTED** [review]

File: app/core/Analytics/MetaMetrics.events.ts (hunk 2 above)
Comment (gantunesr) [reply app/core/Analytics/MetaMetrics.events.ts]: [e1591aa](https://github.com/MetaMask/metamask-mobile/pull/5454/commits/e1591aa1fb2b25dfb7876a995e3e4cef999ae9b2)

File: app/util/analytics.js (hunk 3 above)
Comment (gantunesr) [reply app/util/analytics.js:1]: I thought I had deleted it. Well, anyway [bed860c](https://github.com/MetaMask/metamask-mobile/pull/5454/commits/bed860c517787b08282156341aaa8a8ba778e10b)

**Review by jpcloureiro — APPROVED** [review]

**Pull request merged**